src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,186 @@
+package duke;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+/**
+ * A class that encapsulates the Parser object
+ * which deals with making sense of the user command
+ *
+ * @author  Wee Xin Yang, Markus
+ * @version 0.1
+ * @since   2022-8-24
+ */
+public class Parser {
+
+    private final Ui ui;
+    private SavedTaskHandler storage;
+
+    public Parser(SavedTaskHandler storage) {
+        this.storage = storage;
+        ui = new Ui();
+
+    }
+
+    /**
+     * Returns a boolean value representing whether the input is Numeric
+     * @param string The string that will be parsed to check if it is Numeric
+     *
+     * @return A boolean showing if the input string is Numeric
+     */
+    public static boolean isNumeric(String string) {
+        int intValue;
+
+        if (string == null || string.equals("")) {
+            return false;
+        }
+
+        try {
+            intValue = Integer.parseInt(string);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Parses the user's input and assign different actions according to the input
+     * @param input The string that will be parsed
+     *
+     */
+    public void parse(String input) throws ParseException {
+        String[] splitStr = input.split("\\s+");
+        TaskList taskList = storage.getTaskList();
+
+        if (input.equalsIgnoreCase("bye")) {
+            storage.write(taskList);
+            ui.goodbyeMessage();
+
+        } else if (input.equalsIgnoreCase("list")) {
+            int counter = 1;
+            ui.lineBreak();
+            taskList.list();
+
+        } else if (input.substring(0, Math.min(input.length(), 4)).equals("mark")
+                || input.substring(0, Math.min(input.length(), 6)).equals("unmark")
+                || input.substring(0, Math.min(input.length(), 6)).equals("delete")) {
```

Comment:
Would it be better if these numbers were to define as constants ?  For example:
```
final int START_OF_COMMAND = 0;
final int END_OF_FOUR_WORDS_COMMAND = 4;
```
Then this line `input.substring(0, Math.min(input.length(), 4)).equals("mark")` can be translated to
`input.substring(START_OF_COMMAND , Math.min(input.length(), END_OF_FOUR_WORDS_COMMAND)).equals("mark")`